src/services/TechRecordsService.ts

```diff
@@ -171,7 +170,8 @@ class TechRecordsService {
 
   private validateVrms(techRecord: ITechRecordWrapper) {
     let areVrmsValid = true;
-    if (!techRecord.primaryVrm) {
+    const vehicleType = techRecord.techRecord[0].vehicleType;
+    if ((vehicleType === VEHICLE_TYPE.HGV || vehicleType === VEHICLE_TYPE.PSV) && !techRecord.primaryVrm) {
```

Comment:
here if the vehicle is a trailer (or any other vehicle type) it will go in the else block and it might not work as intended

Reply:
I mean, if the schema doesn't specify it as required, it will say that an empty VRM is valid

Reply:
if the vehicle is an HGV or PSV, primaryVrm is mandatory, for TRL they are optional. So if you don't specify a primaryVrm for an HGV or PSV it should throw an error. If you do specify a primaryVrm it will validate that using Joi.